internal/secrets/encryptor_test.go

```diff
@@ -0,0 +1,285 @@
+	for i := 0; i < 10000; i++ {
+		encrypted, _ := e.EncryptBytes(messageToEncrypt)
+		crypts = append(crypts, string(encrypted))
+	}
+
+	for _, item := range crypts {
+		if isInSliceOnce(item, crypts) == false {
+			t.Fatalf("Duplicate encrypted string found.")
+		}
+	}
+}
+
+func TestKeyMigration(t *testing.T) {
+	keyA, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	keyB, err := generateRandomAESKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	encryptorA := newEncryptor(keyA, nil)
+
+	message := "encrypted with Key A"
+	encryptedMessage, err := encryptorA.EncryptBytes([]byte(message))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// now rotate keys to use Key B
+	encryptorB := newEncryptor(keyB, keyA)
+	decryptedMessage, err := encryptorB.DecryptBytes(encryptedMessage)
+	if err != nil {
+		t.Fatalf("unable to Decrypt string: %v", err)
+	}
+
+	if string(decryptedMessage) != message {
+		t.Fatalf("messages do not match")
+	}
+}
+
+func TestEncryptAndDecryptBytesIfPossible(t *testing.T) {
+	initialKey, _ := generateRandomAESKey()
+	e := newEncryptor(initialKey, nil)
+	encString, err := e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatalf("Failed to EncryptBytes")
+	}
+	if bytes.Equal(encString, messageToEncrypt) {
+		t.Fatalf("Encryption failed.")
+	}
+
+	decString, err := e.DecryptBytes(encString)
+	if err != nil {
+		t.Fatalf("Failed to Decrypt")
+	}
+	if !bytes.Equal(decString, messageToEncrypt) {
+		t.Fatalf("Decryption failed.")
+	}
+
+	// now test when we cannot EncryptBytes
+
+	e = noOpEncryptor{}
+	encString, err = e.EncryptBytes(messageToEncrypt)
+	if err != nil {
+		t.Fatalf("Received error when code path should be nil. %v", err)
+	}
+	if !bytes.Equal(encString, messageToEncrypt) {
+		t.Fatalf("Received encrypted string, expected unencrypted.")
+	}
+	decString, err = e.DecryptBytes(encString)
+	if err != nil {
+		t.Fatalf("Received error when code path should be nil. %v", err)
+	}
+	if !bytes.Equal(decString, messageToEncrypt) {
+		t.Fatalf("Received encrypted string, expected unencrypted.")
+	}
+}
+
+func Test_gatherKeys(t *testing.T) {
+	tests := []struct {
+		name             string
+		data             []byte
+		wantPrimaryKey   []byte
+		wantSecondaryKey []byte
+		wantErr          bool
+	}{
+		{
+			"base-case",
+			[]byte("key123,key345"),
+			[]byte("key123"),
+			[]byte("key345"),
+			false,
+		},
+		{
+			"no key set",
+			[]byte("key123"),
+			[]byte("key123"),
+			nil,
+			false,
+		},
+		{
+			"3 key err case",
+			[]byte("look mom, I am a key, me too"),
+			nil,
+			nil,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotPrimaryKey, gotSecondaryKey, err := gatherKeys(tt.data)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("gatherKeys() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if !bytes.Equal(gotPrimaryKey, tt.wantPrimaryKey) {
+				t.Errorf("gatherKeys() oOldKey = %v, want %v", gotSecondaryKey, tt.wantPrimaryKey)
```

Comment:
You may want to actually print the string to be human readable:
```suggestion
				t.Errorf("gatherKeys() oOldKey = %s, want %s", gotSecondaryKey, tt.wantPrimaryKey)
```